pandas/core/groupby/groupby.py

```diff
@@ -1013,10 +1013,10 @@ def _set_group_selection(self) -> None:
         """
         # This is a no-op for SeriesGroupBy
         grp = self.grouper
-        if not (
-            grp.groupings is not None
-            and self.obj.ndim > 1
-            and self._group_selection is None
+        if (
+            grp.groupings is None
+            or self.obj.ndim == 1
+            or self._group_selection is not None
         ):
```

Comment:
This leaked into here, but I've always found the old version quite hard to read.